pkgs/tools/misc/opentelemetry-collector/releases.nix

```diff
@@ -0,0 +1,172 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+  installShellFiles,
+  testers,
+  nixosTests,
+  opentelemetry-collector-builder,
+  pkgs,
+  go,
+  git,
+  cacert,
+  stdenv,
+}:
+let
+  # This is the tool OTEL uses to build their distributions.
+  builder = "${opentelemetry-collector-builder}/bin/ocb";
+
+  # Keep the version in sync with the builder.
+  rev = opentelemetry-collector-builder.src.rev;
+
+  version = lib.removePrefix "cmd/builder/v" rev;
+
+  # This is a weird meta-repo where all the open-telemetry collectors are.
+  src = fetchFromGitHub {
+    owner = "open-telemetry";
+    repo = "opentelemetry-collector-releases";
+    rev = "v${version}";
+    hash = "sha256-wHi3/rby/AA37x3BCbTXlLgRaN4DpATLSXpQb31Jr/o=";
+  };
+
+  # Then from this src, we use the tool to generate some go code, including
+  # the go.mod and go.sum files.
+  #
+  # The output depends on which release.
+  mkDistributionSource =
+    {
+      name,
+      hash,
+    }:
+    stdenv.mkDerivation {
+      inherit name;
+
+      nativeBuildInputs = [
+        cacert
+        git
+        go
+      ];
+
+      inherit src;
+
+      outputHash = hash;
+      outputHashMode = "recursive";
+      outputHashAlgo = if hash == "" then "sha256" else null;
+
+      patchPhase = ''
+        patchShebangs .
+      '';
+
+      configurePhase = ''
+        export HOME=$NIX_BUILD_TOP/home
+        export GIT_SSL_CAINFO=$NIX_SSL_CERT_FILE
+      '';
+
+      buildPhase = ''
+        # Only generate the go code, skip compilation
+        ./scripts/build.sh -d ${name} -b ${builder} -s true
+      '';
+
+      installPhase = ''
+        mv ./distributions/${name}/_build $out
+
+        # Make it reproducible
+        rm $out/build.log
+      '';
+    };
+
+  # Then, finally, we build the project as a normal go module package.
+  mkDistribution =
+    {
+      name,
+      sourceHash,
+      vendorHash,
+    }:
+    let
+      package = buildGoModule {
+        pname = name;
+        inherit version;
+
+        src = mkDistributionSource {
+          inherit name;
+          hash = sourceHash;
+        };
+
+        inherit vendorHash;
+
+        nativeBuildInputs = [ installShellFiles ];
+
+        # upstream strongly recommends disabling CGO
+        # additionally dependencies have had issues when GCO was enabled that weren't caught upstream
+        # https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md#using-cgo
```

Comment:
This has gotten a stronger warning and was moved to https://github.com/open-telemetry/opentelemetry-collector/blob/9d2685f404ef65b46152ae80c36ff84f303e3cc6/docs/coding-guidelines.md?plain=1#L311 (permalink)

Reply:
side note: systemd journal ingestion seems to still work as the receiver shells out instead of liking against libsystemd, which would require CGO: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/journaldreceiver

Reply:
Can you update the link to the permalink from the first comment?